bundle/src/main/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionJob.java

```diff
@@ -0,0 +1,126 @@
+
+    @Reference
+    Packaging packaging;
+
+    @Reference
+    ResourceResolverFactory resourceResolverFactory;
+
+    @Override
+    public JobResult process(Job job) {
+        Session session = null;
+        String groupName = job.getProperty(GROUP_NAME, String.class);
+        Integer maxAgeInDays = job.getProperty(MAX_AGE_IN_DAYS, Integer.class);
+        LOG.debug("Job Configuration: ["
+                + "Group Name: {}, "
+                + "Service User: {}, "
+                + "Age of Package {} days,]", groupName, SERVICE_USER, maxAgeInDays);
+
+        try (ResourceResolver resourceResolver = resourceResolverFactory.getServiceResourceResolver(
+            Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, SERVICE_USER))) {
+            session = resourceResolver.adaptTo(Session.class);
+            JcrPackageManager packageManager = packaging.getPackageManager(session);
+            List<JcrPackage> packages = packageManager.listPackages(groupName, false);
+
+            for (JcrPackage jcrPackage : packages) {
+                String packageDescription = getPackageDescription(jcrPackage);
+                if (isPackageOldEnough(jcrPackage, maxAgeInDays)) {
+                    packageManager.remove(jcrPackage);
+                    LOG.info("Deleted package {}", packageDescription);
+                } else {
+                    LOG.info("Not removing package because it's not old enough {}", packageDescription);
+                }
+            }
+        } catch (LoginException | RepositoryException | IOException e) {
+            LOG.error("Unable to clear packages due to {}", e.toString());
+            return JobResult.FAILED;
+        } finally {
+            if (session != null) {
+                session.logout();
+            }
+        }
+        LOG.info("Package Garbage Collector job finished");
+        return JobResult.OK;
+    }
+
+    private boolean isPackageOldEnough(JcrPackage jcrPackage, Integer maxAgeInDays) throws RepositoryException, IOException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM);
+        Period maxAge = Period.ofDays(maxAgeInDays);
+        LocalDate oldestAge = LocalDate.now().minus(maxAge);
+        Calendar packageCreatedAtCalendar = jcrPackage.getPackage().getCreated();
+        LocalDate packageCreatedAt = LocalDateTime.ofInstant(
+            packageCreatedAtCalendar.toInstant(),
+            packageCreatedAtCalendar.getTimeZone().toZoneId()).toLocalDate();
+        String packageDescription = getPackageDescription(jcrPackage);
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Checking if package is old enough: Name: {}, Created At: {}, Oldest Age: {}",
+                packageDescription, packageCreatedAt.format(formatter), oldestAge.format(formatter));
+        }
+        return packageCreatedAt.isBefore(oldestAge) || packageCreatedAt.isEqual(oldestAge);
```

Comment:
Not sure if this is easier to understand...
```suggestion
        return !packageCreatedAt.isAfter(oldestAge);
```